ext/curl/multi.c

```diff
@@ -105,28 +110,24 @@ PHP_FUNCTION(curl_multi_add_handle)
 void _php_curl_multi_cleanup_list(void *data) /* {{{ */
 {
 	zval *z_ch = (zval *)data;
-	php_curl *ch;
 
 	if (!z_ch) {
 		return;
 	}
 	if (!Z_RES_P(z_ch)->ptr) {
 		return;
 	}
-	if ((ch = (php_curl *)zend_fetch_resource(Z_RES_P(z_ch), le_curl_name, le_curl)) == NULL) {
-		return;
-	}
 
 	zend_list_delete(Z_RES_P(z_ch));
```

Comment:
I don't understand this: even though that this list deals with resources, if I delete it, more tests fail than before. Should I instead explicitly free objects here? Or what is the problem? Is this list needed at all?

Reply:
I see that this list stored the handles, but it seems that it's not possible to use anymore.

Reply:
zend_lists are all about resources; see http://www.phpinternalsbook.com/php7/internal_types/zend_resources.html for details.

Reply:
Thanks for the link! I certainly read this part in the past, but forgot about it since then :)

So I saw in a comment that this list is for resources only that's why I wasn't sure what data structure to use instead of it. E.g. would a plain old HashTable do it?

Reply:
Probably, but maybe a zend_llist (note the double l) is a reasonable alternative?

Reply:
Hmm, I think I'm starting to get it.. I didn't notice that `zend_llist` and `zend_list` is different kindf of beasts. :) So it turns out that the handles are already stored in a `zend_llist`, and `zend_list_delete()` is only used here because resources can be removed like that.

Reply:
Managed to solve this problem :)